### PR TITLE
feat: reducers-refactor

### PR DESCRIPTION
**Pull request by sorodrigo**
This PR includes:
- Refactors `updater.js` moving reducers logic into a new file `reducers.js`. Reducers are now pure functions. Includes typing into reducer functions.

Flow is still raising errors, but if you don't mind I prefer to address this later.

### CODE REVIEW

File: src/updater.js [hunk 1]
```diff
@@ -1,110 +1,17 @@
...
-
-  if (
-    action.type === OFFLINE_BUSY &&
-    action.payload &&
-    typeof action.payload.busy === 'boolean'
-  ) {
-    return { ...state, busy: action.payload.busy };
-  }
+  const newState =
+    reducers[action.type] && reducers[action.type](state, action);
```
Comment (wacii) [inline src/updater.js]:
```js
if (newState !== null) {
  return newState;
}
```

If you have successfully calculated the new state, you might as well exit the function immediately, as was done before.

**Comment by wacii** [comment]
Unless you have a specific reason otherwise, I would wait until the next version to merge. The reducer wasn't just extracted but refactored, and without tests we can't be 100% sure something hasn't changed.

I think using objects as flow control instead of switch statements will make working with Flow more difficult. Although we're barely using Flow right now, so that might not be much of an issue.

**Comment by sorodrigo** [comment]
Ok, I have no problem on leaving this for next release. And adding tests for the reducer is a good idea.

**Comment by sorodrigo** [comment]
moved to main repo

**Pull request closed without merging**